### PR TITLE
also delete release when deleting tag

### PR DESCRIPTION
**Pull request by rajatjindal**
this attempts to fix issue where sometimes our canary release is marked as 'draft' and re-runs won't help as it will just update the existing release (which is already draft).

I still don't have a good answer as to why the release is getting marked as draft (except someone reporting same issue here https://github.com/ncipollo/release-action/issues/336#issue-1728946267). may be if we push to main while previous release is being tagged?

### CODE REVIEW

**Review by itowlson — APPROVED** [review]
Longer term it would be great to find a better solution, but if this gets us out of the "all CI fails and everything is terrible" hole we're in then :shipit: :shipit: :shipit: and we'll figure it out later!

**Comment by rajatjindal** [comment]
this just happened again (before the merge), so I am going to merge this and monitor if it improves the state.

**Pull request merged**
